source/extensions/common/wasm/wasm.h

```diff
@@ -27,6 +27,16 @@
 #include "include/proxy-wasm/exports.h"
 #include "include/proxy-wasm/wasm.h"
 
+#define RETURN_IF_FALSE(COND, RET_VALUE)                                                           \
```

Comment:
`RETURN_IF_FALSE` is analogous to `RETURN_IF_ERROR` macro, but extended a little bit.

IMHO, there are several reason to use this kind of macro.
1. This may help to make the codes clean, because we can focus on the main logic, not error handling. In addition, like `ASSERT`, we can regard `COND` of `RETURN_IF_FALSE` as an invariant at the place.
2. As you can see, I used it for the error handling codes which happen very rarely. The coverage for such lines are not helpful to understand which parts need to add test cases.

Reply:
Yeah. frankly, it may be seen as a "cheating".

I think that the interface restructuring may not work well. (I can't imagine how to restructure it. If you have any idea, please let me know.)

IMO, if we cannot accept this macro, I think we need to make some artificial mock (celValue mock) to raise the rare error cases. That is because, for example, `SerializeToString` return `false` when the length of the serialized string is over 2GB. To test just a test case, generating 2GB data in memory would be not a good choice.

Reply:
Oh.. the "mock" above is for [context.cc](https://github.com/envoyproxy/envoy/pull/23055/files#diff-c386f59113c79157fdc6b88ce0c4c14e307509bd7038ce816821985293d55410R373-R408).

Reply:
> 2. As you can see, I used it for the error handling codes which happen very rarely. The coverage for such lines is not helpful to understand which parts need to add test cases.

I would argue that this might be good to have explicit tests that validate these paths, just to ensure that these tests catch if anything changes in the underlying library functions or an invariant no longer holds.

Reply:
Remove the macros and made some tests more.